editor/editor_properties_array_dict.cpp

```diff
@@ -867,14 +868,15 @@ void EditorPropertyDictionary::setup(PropertyHint p_hint) {
 void EditorPropertyDictionary::update_property() {
 	Variant updated_val = get_edited_property_value();
 
-	if (updated_val.get_type() == Variant::NIL) {
+	if (updated_val.get_type() != Variant::DICTIONARY) {
```

Comment:
this is to not only check for `null` but for anything that wouldn't be treated well (only dictionary is expected) just as `EditorPropertyArray` checks with `is_array`.